pkg/productinfo/azure/productinfo_azure.go

```diff
@@ -345,15 +351,46 @@ func (a *AzureInfoer) GetZones(region string) ([]string, error) {
 
 // GetRegions returns a map with available regions transforms the api representation into a "plain" map
 func (a *AzureInfoer) GetRegions() (map[string]string, error) {
-	regions := make(map[string]string)
-	locations, err := a.subscriptionsClient.ListLocations(context.TODO(), a.subscriptionId)
-	if err != nil {
+
+	allLocations := make(map[string]string)
+	supLocations := make(map[string]string)
+
+	// retrieve all locations for the subscription id (some of them may not be supported by the required provider
+	if locations, err := a.subscriptionsClient.ListLocations(context.TODO(), a.subscriptionId); err == nil {
+		// fill up the map: DisplayName - > Name
+		for _, loc := range *locations.Value {
+			allLocations[*loc.DisplayName] = *loc.Name
+		}
+	} else {
+		log.Errorf("error while retrieving azure locations. err: %s", err.Error())
 		return nil, err
 	}
-	for _, loc := range *locations.Value {
-		regions[*loc.Name] = *loc.DisplayName
+
+	// identify supported locations for the namespace and resource type
+	const (
+		providerNamespace = "Microsoft.Compute"
+		resourceType      = "locations/vmSizes"
+	)
+
+	if providers, err := a.providersClient.Get(context.TODO(), providerNamespace, ""); err == nil {
+		for _, pr := range *providers.ResourceTypes {
+			if *pr.ResourceType == resourceType {
+				for _, displName := range *pr.Locations {
+					if loc, ok := allLocations[displName]; ok {
+						log.Debugf("found supported location. [name, display name] = [%s, %s]", loc, displName)
+						supLocations[loc] = loc
```

Comment:
this should be `supLocations[loc] = displName` to store the display name as well

Reply:
thanks, that was the intention :)